tiledb/sm/filter/float_scaling_filter.cc

```diff
@@ -87,6 +90,19 @@ Status FloatScalingFilter::run_forward(
     const T* part_data = static_cast<const T*>(i.data());
     for (uint32_t j = 0; j < num_elems_in_part; ++j) {
       T elem = part_data[j];
+
+      // We should only handle numbers that are either normalized or 0.
+      switch (std::fpclassify(elem)) {
```

Comment:
@abigalekim this check needs to happen after the computation, and we need to check that the classification of the input matches the classification of the output.

Reply:
(per @eric-hughes-tiledb -- we also need to check for integer overflow, if not done already)

Reply:
We are addressing overflow in a different PR.

Reply:
Wait, also how? We shouldn't store the input data, and since we're converting and storing as integers, it doesn't really make sense to support infinite/NaN numbers.

Reply:
Check `fpclassify(elem) == fpclassify( [the floating point expression inside round ] )`

Reply:
because of this change, zeros and denormalized array tests fail. is this expected? should I make the change in this PR? There are other PRs coming up to address this specific change.